### PR TITLE
[tests] remove deprecated tests for model loading

### PR DESCRIPTION
**Pull request by faaany**
## What does this PR do?
2 tests keep failing in `test_modeling_utils.py`:
```bash
FAILED tests/test_modeling_utils.py::ModelUtilsTest::test_legacy_load_from_url - huggingface_hub.utils._validators.HFValidationError: Repo id must be in the form 'repo_name' or 'namespace/repo_name': 'https://huggingf...
FAILED tests/test_modeling_utils.py::ModelUtilsTest::test_load_from_one_file - huggingface_hub.utils._validators.HFValidationError: Repo id must be in the form 'repo_name' or 'namespace/repo_name': '/tmp/tmpoa14so_e...
```
The reason is that `https://huggingface.co/hf-internal-testing/tiny-random-bert/resolve/main/pytorch_model.bin` and `/tmp/xxx` are not valid `repo_id`s. Currently, only `path to a local folder` or `repo_id of a model on the Hub` are accepted. 

Below is the detailed error message:
```bash
E           huggingface_hub.utils._validators.HFValidationError: Repo id must be in the form 'repo_name' or 'namespace/repo_name': 'https://huggingface.co/hf-internal-testing/tiny-random-bert/resolve/main/pytorch_model.bin'. Use `repo_type` argument if needed.
/home/fanli/.conda/envs/dev/lib/python3.10/site-packages/huggingface_hub/utils/_validators.py:158: HFValidationError
```

In my opinion, this is technical debt and we should only test up-to-date features. So I remove them instead of modifying them. What do you think? @ArthurZucker @ydshieh

### CODE REVIEW

**Comment by faaany** [comment]
Hi @ydshieh, could you help review this PR? In my opinion, these 2 tests are just technical debts and should be cleaned-up.

**Comment by ydshieh** [comment]
Hi @faaany Thank you for flagging this situation and opening a PR. I am up to remove these 2 tests, but on our CI history, I saw they started to fail on August 18 2023. However, I can't see any change in `huggingface_hub` at/around that day.

So I would like to ping @Wauplin to see if he remembered something during that time period.

**Comment by Wauplin** [comment]
@ydshieh can't tell what changed on August 12th 2023 specifically on this. The `huggingface_hub.utils._validators.HFValidationError` is expected if an incorrect repo_id is passed (typically a URL or a path) to any `huggingface_hub` method. But this validator dates back to August/Sept 2022 so quite a long time before. I would say most likely how things are handled internally in `transformers` have changed and a call to `huggingface_hub` is now made without verifying if it's a path/url or not.

I'm fine with removing completely the tests but then it means the feature must be considered as removed from `transformers`, not just deprecated until v5. For the end user it doesn't change anything since the feature is broken anyway. (by feature I mean "load from single file or URL"). => it would be good to check if there is any mention to this or any logic in `transformers` related to this feature and remove them altogether so that we don't keep dead code in the main source code.

**Comment by ydshieh** [comment]
Thank you @Wauplin for the comments. Good points.

**Review by ydshieh — APPROVED** [review]
LGTM, thank you again!

File: tests/test_modeling_utils.py [hunk 1]
```diff
@@ -880,26 +879,6 @@ def test_cached_files_are_used_when_internet_is_down(self):
             # This check we did call the fake head request
             mock_head.assert_called()
 
-    def test_load_from_one_file(self):
```
Comment (amyeroberts) [inline tests/test_modeling_utils.py:883]: The equivalent should also be removed from `test_modeling_tf_utils.py`

File: tests/test_modeling_utils.py [hunk 2]
```diff
@@ -880,26 +879,6 @@ def test_cached_files_are_used_when_internet_is_down(self):
...
-                http_get(
-                    "https://huggingface.co/hf-internal-testing/tiny-random-bert/resolve/main/pytorch_model.bin", f
-                )
-
-            config = BertConfig.from_pretrained("hf-internal-testing/tiny-random-bert")
-            _ = BertModel.from_pretrained(tmp_file, config=config)
-        finally:
-            os.remove(tmp_file)
-
-    def test_legacy_load_from_url(self):
```
Comment (amyeroberts) [inline tests/test_modeling_utils.py:896]: Other implementations of this test need to be removed in the repo too e.g. in `test_image_processing_utils.py`

**Review by amyeroberts — COMMENTED** [review]
Thanks for working on this @faaany! 

We should extend these deletions to all the function equivalents in the test suite.

Happy for them to be removed as they're currently not working, and this isn't a feature I've seen anyone complain about not working either.

**Comment by faaany** [comment]
@amyeroberts thanks for the review! I removed the equivalent tests from `test_modeling_tf_utils.py` and `test_image_processing_utils.py`.

**Review by amyeroberts — APPROVED** [review]
Awesome - thanks for this tidy up! 

Could you try rebasing on main? I think this should resolve the currently failing (unrelated) test

**Comment by faaany** [comment]
> Awesome - thanks for this tidy up!
> 
> Could you try rebasing on main? I think this should resolve the currently failing (unrelated) test

done!

**Pull request merged**